nrf_security/Kconfig

```diff
@@ -350,99 +493,219 @@ config VANILLA_MBEDTLS_DHM_C
+	select CC310_MBEDTLS_ECJPAKE_C if CC310_BACKEND
+	default y
 	help
-	  Enable the ECDSA module from mbed TLS vanilla.
-	  MBEDTLS_ECDSA_C setting in mbed TLS config file.
-endmenu
+	  Enable support for ECJPAKE
+
+menu "ECC curves"
+	depends on (CC310_BACKEND || MBEDTLS_VANILLA_BACKEND)
+
+config MBEDTLS_ECP_DP_SECP192R1_ENABLED
+	bool "Enable NIST curve secp192r1"
+	default n
+	help
+	  MBEDTLS_ECP_DP_SECP192R1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_SECP224R1_ENABLED
+	bool "Enable NIST curve secp224r1"
+	default n
+	help
+	  MBEDTLS_ECP_DP_SECP224R1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_SECP256R1_ENABLED
+	bool "Enable NIST curve secp256r1"
+	default y
+	help
+	  MBEDTLS_ECP_DP_SECP256R1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_SECP384R1_ENABLED
+	bool "Enable NIST curve secp384r1"
+	default n
+	help
+	  MBEDTLS_ECP_DP_SECP384R1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_SECP521R1_ENABLED
+	bool "Enable NIST curve secp521r1"
+	default n
+	help
+	  MBEDTLS_ECP_DP_SECP521R1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_SECP192K1_ENABLED
+	bool "Enable Koblitz curve secp192k1"
+	default n
+	help
+	  MBEDTLS_ECP_DP_SECP192K1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_SECP224K1_ENABLED
+	bool "Enable Koblitz curve secp224k1"
+	default n
+	help
+	  MBEDTLS_ECP_DP_SECP224K1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_SECP256K1_ENABLED
+	bool "Enable Koblitz curve secp256k1"
+	default n
+	help
+	  MBEDTLS_ECP_DP_SECP256K1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_BP256R1_ENABLED
+	bool "Enable Brainpool curve bp256r1"
+	depends on (VANILLA_MBEDTLS_ECDSA_C || VANILLA_MBEDTLS_ECP_C || VANILLA_MBEDTLS_ECDH_C  )
+	default n
+	help
+	  MBEDTLS_ECP_DP_BP256R1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_BP384R1_ENABLED
+	bool "Enable Brainpool curve bp384r1"
+	depends on (VANILLA_MBEDTLS_ECDSA_C || VANILLA_MBEDTLS_ECP_C || VANILLA_MBEDTLS_ECDH_C  )
+	default n
+	help
+	  MBEDTLS_ECP_DP_BP384R1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_BP512R1_ENABLED
+	bool "Enable Brainpool curve bp384r1"
+	depends on (VANILLA_MBEDTLS_ECDSA_C || VANILLA_MBEDTLS_ECP_C || VANILLA_MBEDTLS_ECDH_C  )
+	default n
+	help
+	  MBEDTLS_ECP_DP_BP512R1_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_CURVE25519_ENABLED
+	bool "Enable Curve25519 aka Edwards25519"
+	default n
+	help
+	  MBEDTLS_ECP_DP_CURVE25519_ENABLED setting in mbed TLS config file.
+
+config MBEDTLS_ECP_DP_CURVE448_ENABLED
+	bool "Enable Curve448 aka Edwards448"
+	depends on (VANILLA_MBEDTLS_ECDSA_C || VANILLA_MBEDTLS_ECP_C || VANILLA_MBEDTLS_ECDH_C	)
+	default n
+	help
+	  MBEDTLS_ECP_DP_CURVE448_ENABLED setting in mbed TLS config file.
+
+endmenu # ECC curves
+
+endmenu # ECC - Elliptic Curve Cryptography
+
+menu "RSA   - Rivest–Shamir–Adleman cryptosystem"
```

Comment:
I think changing this from menuconfig to menu is a step back in usability, as users must now enter the submenu to see if the group is configured or not.
The menuconfig allows user to completely disable RSA if it is not needed.